### ISSUE TITLE
The localization system is always in English

### ISSUE THREAD

**Issue by zqzcsilver** [bluemagic123/tModLoader#3381]
### Version

Latest 1.4.4-preview

### OS

Windows

### Platform

Steam

### This bug affects

Mod capability as a Modder

### Description

I modified my hjson format according to ExampleMod, and it didn't make errors in reading, but its text always showed hjson keys instead of values in the game. I used dnSpy to debug tModLoader and found that it always has en-US parameters when loading Mod's hjson, and it is not affected by the game settings. I think this is a very weird bug and hope to fix it soon.

**Comment by Chicken-Bones** [comment]
Please provide a zip or github link to your mod

**Comment by zqzcsilver** [comment]
https://github.com/zqzcsilver/OdeMod

**Comment by zqzcsilver** [comment]
![image](https://user-images.githubusercontent.com/60340270/229883825-3335cfdc-a887-4c16-b4e6-735637416300.png)
![image](https://user-images.githubusercontent.com/60340270/229883920-0efb934e-7115-4bbe-afb6-9ece713053c3.png)

**Comment by Chicken-Bones** [comment]
The localization system loads english and then loads any other selected language _over the top_. 

Yes, an english localization file is required, to specify which keys are valid. The localization exporter should automatically produce one for you the first time your mod is run, provided your mod sources are in the `ModSources` folder. 

There is no need for you to fill out the english localization file, but it must be present with all keys defined.

**Comment by zqzcsilver** [comment]
My mod was placed in the `ModSource` folder, but tModLoader didn't generate a localization file for me in English.

**Comment by Chicken-Bones** [comment]
@zqzcsilver can you come to Discord? Perhaps we can help you there. Also please provide a `client.log`

**Comment by zqzcsilver** [comment]
If you're sure it's not an issue with tModLoader, I can fix it myself by debugging. I was just worried that other modders would encounter the same issues and bugs as me, so I brought up this issue. Thank you very much for your help and answers.If you confirm that this is not an issue caused by tModLoader not running as expected, then I can close the issue.

**Comment by Chicken-Bones** [comment]
I downloaded your mod to test, and I found that you have a bit of a chicken-egg situation. The exporter won't update the english localization file, because there isn't one so it doesn't think you want english as a language, but then the mod loader won't load any of your localizations because english is the 'definition' language for all the keys.

You can work around this by simply creating an empty `en-US.hjson` file next to you `zh-Hans.hjson` and building and reloading your mod.

@JavidPack this is a problem with an unintuitive workaround atm. Ideas:
- The exporter could always produce english files, but I'm not sure how this might interact with the translation mod feature etc
- The localization loader could use another language other than english as the 'root' language to get keys from in `Autoload` (but which)

I'll leave this decision to you

**Comment by zqzcsilver** [comment]
Yes, after adding en-US.hjson, the localization system returned to normal. But I think that the localization system should still load the game text in the corresponding language even if the corresponding language has the game text, rather than loading the text in other languages only after having en-US.hjson, which is very confusing. In the absence of English text, if the game is set to English, the text key should be used to display the text. Finally, thank you very much for your help, I think this issue should be closed.
